storage/series/buffer.go

```diff
@@ -124,18 +123,18 @@ type databaseBufferDrainFn func(b block.DatabaseBlock)
 
 func newDatabaseBuffer(drainFn databaseBufferDrainFn, opts Options) databaseBuffer {
 	b := &dbBuffer{
-		opts:    opts,
-		nowFn:   opts.ClockOptions().NowFn(),
 		drainFn: drainFn,
 	}
-	b.Reset(opts.RetentionOptions())
+	b.Reset(opts)
 	return b
 }
 
-func (b *dbBuffer) Reset(ropts retention.Options) {
-	b.blockSize = ropts.BlockSize()
-	b.bufferPast = ropts.BufferPast()
-	b.bufferFuture = ropts.BufferFuture()
+func (b *dbBuffer) Reset(opts Options) {
+	b.opts = opts
+	b.nowFn = opts.ClockOptions().NowFn()
+	b.blockSize = opts.RetentionOptions().BlockSize()
+	b.bufferPast = opts.RetentionOptions().BufferPast()
+	b.bufferFuture = opts.RetentionOptions().BufferFuture()
```

Comment:
nit: Slightly more optimal (and considering this code path is executed a lot during series creation):
```go
retentionOpts := opts.RetentionOptions()
b.blockSize = retentionOpts.BlockSize()
// .. same for bufferPast and bufferFuture
```